source/sgp_mode/DiversityAnalysis.h

```diff
@@ -30,85 +30,182 @@
 #include <math.h>
 #include <set>
 
-
+/**
+ *
+ * Input: A cpu of an organism
+ *
+ * Output: A length 64 emp bitset which describes the phenotype of organism
+ * such that the ith 1 in the bitset marks the completion of task i.
+ *
+ * Purpose: Get the phenotype of an organism
+ *
+ */
 emp::BitSet<64> ReturnTasksDone(CPU org_cpu) {
   
   org_cpu.RunCPUStep(emp::WorldPosition::invalid_id, 100);
   
-
   return *org_cpu.state.used_resources;
 }
 
 
-
-
-emp::unordered_map<emp::BitSet<64>, int> GetPhenotypeMap(emp::vector<CPU> organisms){ // SGPWorld world
-//pointer??
-    emp::unordered_map<emp::BitSet<64> , int> phenotype;
+/**
+ *
+ * Input: A vector of cpus of organisms
+ *
+ * Output: A map with keys being phenotypes and values being counts of organisms
+ * with corresponding phenotypes
+ *
+ * Purpose: Get the count of phenotypes
+ *
+ */
+emp::unordered_map<emp::BitSet<64>, int> GetPhenotypeMap(emp::vector<CPU> organisms){
+    emp::unordered_map<emp::BitSet<64> , int> phenotype_counts;
     for (size_t i = 0; i < organisms.size(); i++){
-        CPU org_cpu = organisms[i];//.GetCPU();//??
-        emp::BitSet<64> managed_tasks = ReturnTasksDone(org_cpu);
-        
-        // std::cout<<"got phenotype: " << managed_tasks << std::endl; 
-        phenotype[managed_tasks] += 1; // this works with / without key exists (i.e. just like python)
-        
+        CPU org_cpu = organisms[i];
+        emp::BitSet<64> managed_tasks = ReturnTasksDone(org_cpu);        
+        if (phenotype_counts.count(managed_tasks)){
+            phenotype_counts[managed_tasks] += 1; 
+        } else {
+            phenotype_counts[managed_tasks] = 1; //necessary initializing for c++ maps
+        }
     }
-    return phenotype;
+    return phenotype_counts;
+}
 
 
-}
 
 
+/**
+ *
+ * Input: A vector of cpus of organisms
+ *
+ * Output: A map with keys being phenotypes and values being counts of organisms
+ * with corresponding phenotypes
+ *
+ * Purpose: Get the count of phenotypes
+ *
+ */
+emp::vector<emp::BitSet<64>> GetPhenotypeVector(emp::vector<CPU> organisms){
+    emp::vector<emp::BitSet<64>> phenotypes;
+    for (size_t i = 0; i < organisms.size(); i++){
+        CPU org_cpu = organisms[i];
+        emp::BitSet<64> managed_tasks = ReturnTasksDone(org_cpu);        
+        phenotypes.push_back(managed_tasks);
+    }
+    return phenotypes;
+}
 
-double AlphaDiversityHelper(emp::unordered_map<emp::BitSet<64> , int> phenotype, double q = 0.0) {
-    int pop = 0;
+/**
+ *
+ * Input: A map containing the count of phenotypes
+ *
+ * Output: The alpha diversity of the population
+ *
+ * Purpose: Calculates the alpha diversity based on the formula
+ *
+ */
+double AlphaDiversityHelper(emp::vector<emp::BitSet<64>> phenotypes, int j = 4, double q = 0.0) {
```

Comment:
You should include the optional parameters j and q in the documentation for what they are